In [2]:
from scipy import stats
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

In [9]:
data = pd.read_csv('hw_25000.csv', delimiter = ';', names = ['index', 'height_inches', 'weight_pounds'], header = 0, skiprows = 1)

In [17]:
data['height'] = data['height_inches'] * 2.54

In [20]:
data_sample = data[data['height']>=170].merge(data[data['height']<=180]).head(20)

In [30]:
sample_w = data_sample['weight_pounds'].tolist()

In [31]:
data_w = data['weight_pounds'].tolist()

In [32]:
stats.ttest_ind(sample_w, data_w)

Ttest_indResult(statistic=2.2406349413640894, pvalue=0.02505847065887213)

## Так как pvalue=0.025 и это меньше 0.05, делаем вывод что разница статистически значима.